Signal/src/ViewControllers/NewGroupViewController.m

```diff
@@ -340,254 +463,131 @@ - (void)createGroup
-    [actionSheetController addAction:dismissAction];
-
-    UIAlertAction *takePictureAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"MEDIA_FROM_CAMERA_BUTTON", @"media picker option to take photo or video")
-                                                                style:UIAlertActionStyleDefault
-                                                              handler:^(UIAlertAction * _Nonnull action) {
-                                                                  [self takePicture];
-                                                              }];
-    [actionSheetController addAction:takePictureAction];
-
-    UIAlertAction *choosePictureAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"MEDIA_FROM_LIBRARY_BUTTON", @"media picker option to choose from library")
-                                                                  style:UIAlertActionStyleDefault
-                                                                handler:^(UIAlertAction * _Nonnull action) {
-                                                                    [self chooseFromLibrary];
-                                                                }];
-    [actionSheetController addAction:choosePictureAction];
-
-    [self presentViewController:actionSheetController animated:true completion:nil];
-}
+- (void)setGroupAvatar:(nullable UIImage *)groupAvatar
+{
+    OWSAssert([NSThread isMainThread]);
 
-#pragma mark - Group Image
+    _groupAvatar = groupAvatar;
 
-- (void)takePicture {
-    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
-    picker.delegate                 = self;
-    picker.allowsEditing            = NO;
-    picker.sourceType               = UIImagePickerControllerSourceTypeCamera;
+    self.hasUnsavedChanges = YES;
 
-    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
-        picker.mediaTypes = [[NSArray alloc] initWithObjects:(NSString *)kUTTypeImage, nil];
-        [self presentViewController:picker animated:YES completion:[UIUtil modalCompletionBlock]];
-    }
+    [self updateAvatarView];
 }
 
-- (void)chooseFromLibrary {
-    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
-    picker.delegate                 = self;
-    picker.sourceType               = UIImagePickerControllerSourceTypeSavedPhotosAlbum;
+- (void)updateAvatarView
+{
+    UIImage *image = (self.groupAvatar ?: [UIImage imageNamed:@"empty-group-avatar"]);
+    OWSAssert(image);
 
-    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeSavedPhotosAlbum]) {
-        picker.mediaTypes = [[NSArray alloc] initWithObjects:(NSString *)kUTTypeImage, nil];
-        [self presentViewController:picker animated:YES completion:[UIUtil modalCompletionBlock]];
-    }
+    self.avatarView.image = image;
+    self.avatarView.layer.borderColor = [[UIColor lightGrayColor] CGColor];
```

Comment:
Yes, I was wondering the same thing. It definitely looks better for avatars with large white portions. I've added a card.